configure.py

```diff
@@ -2402,5 +2403,56 @@ def create_building_system(args):
     generate_compdb('compile_commands.json', ninja, args.buildfile, selected_modes)
 
 
+def configure_for_mode(mode, args):
+    # TODO: use Ninja multi-config
+    source_dir = os.path.realpath(os.path.dirname(__file__))
+    build_dir = os.path.join(source_dir, 'build', mode)
+
+    settings = {
+        'CMAKE_BUILD_TYPE': mode,
+        'CMAKE_CXX_COMPILER': args.cxx,
+        'CMAKE_C_COMPILER': args.cc,
+        'CMAKE_C_FLAGS': args.user_cflags,
+        'CMAKE_EXE_LINKER_FLAGS': semicolon_separated(args.user_ldflags),
+        'CMAKE_EXPORT_COMPILE_COMMANDS': 'ON',
+        'CMAKE_EXPORT_NO_PACKAGE_REGISTRY': 'ON',
+    }
+    if args.date_stamp:
+        settings['Scylla_DATE_STAMP'] = args.date_stamp
+
+    cmake_command = ['cmake']
+    cmake_command += [f'-D{var}={value}' for var, value in settings.items()]
+    cmake_command += ['-G', 'Ninja',
+                      '-B', build_dir,
+                      '-S', source_dir]
+    subprocess.check_call(cmake_command, shell=False, cwd=source_dir)
+    return build_dir, os.path.join(build_dir, 'build.ninja')
+
+
+def configure_with_cmake(args):
+    ninja = find_ninja()
+    with open('build.ninja', 'w', encoding='utf-8') as main_build_file:
+        ninja_opts = ''
+        if args.verbose:
+            ninja_opts += ' --verbose'
+        main_build_file.write(textwrap.dedent(f'''
+          rule ninja
+            command = {ninja} {ninja_opts} -C $dir $target
+            restat = 1
+        '''))
+
+        all_modes = ['debug', 'release', 'dev', 'sanitize', 'coverage']
+        for mode in args.selected_modes or all_modes:
+            build_dir, build_file = configure_for_mode(mode.capitalize(), args)
+            main_build_file.write(textwrap.dedent(f'''
+              build {mode}-build: ninja {build_file}
+                dir = {build_dir}
+                target = scylla
```

Comment:
Please add more compatibility targets (e.g. dist stuff, dev-headers, and also the default target and other mode-independent targets) to reduce friction. In particular, any target that Jenkins uses.

Reply:
sure. i planned to do this in multiple iterations. will add them in the same PR.

Reply:
actually, i am not sure this is the right approach to add per-mode targets using a driver `build.ninja`. probably a better way to do this is to use the multi-config generator and `CMAKE_CROSS_CONFIGS`. but before we support multi-config generator, we need to support it in Seastar first.

Reply:
i start working on the Seastar's side. see https://github.com/scylladb/seastar/pull/1919

Reply:
A bit late, but I agree the multi-config stuff is what we want.